### ISSUE TITLE
pdf-tools documentation in need of an update?

### ISSUE THREAD

**Issue by 88-percent** [syl20bnr/spacemacs#14791]
The readme document here appears to have some issues when discussing the install of pdf-tools at #2:
https://develop.spacemacs.org/layers/+tools/pdf-tools/README.html

1) the document is silent as to adding "pdf" to the configuration layers
  - I'd wrongly guessed that I should have added "pdf-tools" 
2) the document suggests adding the line "(setq-default dotspacemacs-configuration-layers '(pdf-tools))" when it looks like it should read "(setq-default dotspacemacs-configuration-layers '(pdf))"
3) The document is confusing as to the different steps needed for OSX, Debian based distros and other distros. The following lines make it difficult to understand which steps should be skipped:
"Note that compiling from source might not be necessary" - does this mean that compiling from source may be necessary on Debian based distros? Is this referring to OSX only?
"If you use a Debian based system such as Ubuntu, you can skip the step" - assumed this meant installing the dependencies manually, but it could also refer to compiling from source. 
"After installing the dependencies, add this to your ~/.spacemacs" - again, which OS/distro needs to have the dependencies installed manually?

A potential solution might be a single line that MacOS is not officially supported and then detailing the install instructions into Debian installs and non-Debian installs.

I was eventually able to get this working with some help from the chat.

**Comment by lebensterben** [comment]
> The readme document here appears to have some issues when discussing the install of pdf-tools at #2:
> 
> https://develop.spacemacs.org/layers/+tools/pdf-tools/README.html
> 
> 
> 
> 1) the document is silent as to adding "pdf" to the configuration layers
> 
>   - I'd wrongly guessed that I should have added "pdf-tools" 

This link is outdated and it should be https://develop.spacemacs.org/layers/+readers/pdf/README.html

Note that this outdated link is not presented in layers list.

How do you get there?


> 2) the document suggests adding the line "(setq-default dotspacemacs-configuration-layers '(pdf-tools))" when it looks like it should read "(setq-default dotspacemacs-configuration-layers '(pdf))"

In the correct layer documentation it's correctly specified.

> 3) The document is confusing as to the different steps needed for OSX, Debian based distros and other distros. The following lines make it difficult to understand which steps should be skipped:
> 
> "Note that compiling from source might not be necessary" - does this mean that compiling from source may be necessary on Debian based distros? Is this referring to OSX only?

Read the full paragraph: "You'll need to install a few libraries. Check the instructions on pdf-tools page. Note that compiling from source might not be necessary, as the documentation states."

It's not ambiguous if you follow the instructions.

> "If you use a Debian based system such as Ubuntu, you can skip the step" - assumed this meant installing the dependencies manually, but it could also refer to compiling from source. 

Again read the full paragraph: "If you use a Debian based system such as Ubuntu, you can skip this step, as pdf-tools installation asks whether you want it to try to download automatically all dependencies using apt-get."

It clearly is talking about on Debian based systems pdf-tools can download and install dependencies automatically with apt-get, so that you don't need to manually build dependencies.

> "After installing the dependencies, add this to your ~/.spacemacs" - again, which OS/distro needs to have the dependencies installed manually?

No. This mentioned nothing about host OS or the way you installed the dependencies. It simply means after you installed dependencies, either manually or via a package manager, regardless of the host OS, add "pdf" to your layers list.

> A potential solution might be a single line that MacOS is not officially supported and then detailing the install instructions into Debian installs and non-Debian installs.

> I was eventually able to get this working with some help from the chat.

- To sum up, we need to remove the outdated documentation. But importantly, **how do you get there**?, as this doesn't seem to be linked anywhere.
- I think we should just remove any instructions on installation. User should follow the upstream instructions.

**Comment by HanshenWang** [comment]
He wasn't the only one. This exact same scenario happened to me when I first
started Spacemacs. It could very well be that we are both idiots.

Lol, anyways, google "spacemacs pdf layer" and the 2016 documentation is the 2nd
hit.

https://www.spacemacs.org/layers/+tools/pdf-tools/README.html

When I first tried a year ago, that link was probably the 1st hit. In addition,
if you look at the commit that changes this,
https://github.com/syl20bnr/spacemacs/commit/b37e6d687c331a4b71b1ca2b2a7d77ba73e04dc3,
another user was burned by outdated documentation. The OP probably found that
outdated link first (whichever one), and one cannot, in all fairness, expect a new user to know
this.

Of course, I've hopelessly sold part of my soul to emacs now, and so I'm very
much in your debt @lebensterben. @88-percent don't hesitate to spend a little
extra effort to submit a pull request fixing what exactly wasn't clear from
the docs. Pay it forward!

Speaking of which, the upstream docs link is to the old pdf-tools package. I've just
pointed it to the new repository.

A symptom of a more serious issue I'd say. What to do with master branch and the old
spacemacs.org. I'll keep an eye out to help whichever way I can.

Cheers,

**Comment by 88-percent** [comment]
We all owe the devs like Lucius a debt of gratitude. Spacemacs is allowing me to do things with emacs I couldn't have dreamed of doing with my own vanilla install.

The link I referenced was the first hit on DuckDuckGo's search for: spacemacs pdf-tools. It looked official, so I figured the mistake had been mine, which I guess after all it was.

I'd only posted here trying to help out. I had no idea the mistake was mine.

I'd love to be able to contribute to documentation or tutorials. I don't code, so I'd imagine that's the extent I could pay it forward and help the community. That said, it looks there's alot I need to learn yet. So may be best that I RTFM before I try and make any more suggestions. 

On that note would someone mind being so kind as to confirm that this link is the most up-to-date documentation?

https://www.spacemacs.org/doc/DOCUMENTATION.html

Last thing I need to do is make this worse. Thanks so much everyone.

**Comment by lebensterben** [comment]
it's not

https://develop.spacemacs.org/doc/DOCUMENTATION.html

**Comment by 88-percent** [comment]
Thanks.
